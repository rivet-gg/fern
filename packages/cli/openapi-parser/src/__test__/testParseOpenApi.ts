import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse, SpecImportSettings } from "../parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(
    fixtureName: string,
    openApiFilename: string,
    asyncApiFilename?: string,
    settings?: SpecImportSettings
): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;
            const specs = [];
            if (absolutePathToOpenAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToOpenAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
