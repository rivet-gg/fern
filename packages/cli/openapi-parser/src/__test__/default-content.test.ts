import { testParseOpenAPI } from "./testParseOpenApi";

describe("open api parser", () => {
    testParseOpenAPI("default-content", "openapi.yml");
});
