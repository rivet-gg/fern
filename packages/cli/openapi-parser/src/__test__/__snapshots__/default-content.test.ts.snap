// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open api parser default-content parse open api 1`] = `
{
  "apiVersion": null,
  "basePath": null,
  "channel": [],
  "description": null,
  "endpoints": [
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [],
      "generatedRequestName": "GenerateContentRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "POST",
      "operationId": "GenerateContent",
      "pagination": null,
      "path": "/generateContent",
      "pathParameters": [
        {
          "availability": null,
          "description": null,
          "name": "model",
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GenerateContentRequestModel",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": null,
              "format": null,
              "maxLength": null,
              "minLength": null,
              "pattern": null,
              "type": "string",
            },
            "type": "primitive",
          },
          "variableReference": null,
        },
      ],
      "queryParameters": [],
      "request": null,
      "requestNameOverride": null,
      "response": {
        "description": "Successful operation",
        "responseProperty": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "GenerateContentResponse",
          "groupName": null,
          "nameOverride": null,
          "schema": "GenerateContentResponse",
          "type": "reference",
        },
        "type": "json",
      },
      "sdkName": null,
      "server": [],
      "summary": null,
      "tags": [
        "generativelanguage",
      ],
    },
  ],
  "globalHeaders": [],
  "groups": {},
  "hasEndpointsMarkedInternal": false,
  "idempotencyHeaders": [],
  "nonRequestReferencedSchemas": [
    "GenerateContentResponse",
  ],
  "schemas": {},
  "securitySchemes": {},
  "servers": [
    {
      "description": null,
      "name": null,
      "url": "https://ai.com",
    },
  ],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Default Content Type",
  "variables": {},
  "webhooks": [],
}
`;
